File: 백준/Bronze/2979. 트럭 주차/트럭 주차.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;

		st = new StringTokenizer(br.readLine());
		int A = Integer.parseInt(st.nextToken());
		int B = Integer.parseInt(st.nextToken()) * 2;
		int C = Integer.parseInt(st.nextToken()) * 3;
		int[] score = new int[] {0,A,B,C};
		
		int price = 0;
		int[] time = new int[101];
		for (int i = 0; i < 3; i++) {
			st = new StringTokenizer(br.readLine());
			int sta = Integer.parseInt(st.nextToken());
			int end = Integer.parseInt(st.nextToken());
			
			for (int j = sta; j < end; j++) {
				time[j]++;
			}
		}
		for (int i = 0; i < 101; i++) {
			price += score[time[i]];
		}
		System.out.println(price);
        br.close();
	}
}

File: 백준/Bronze/2979. 트럭 주차/트럭 주차.py
import sys
input = sys.stdin.readline

A,B,C = map(int, input().split())
B *= 2
C *= 3
price = 0
time = [0 for _ in range(101)]

for i in range(3) :
    sta, end = map(int, input().split())
    for j in range(sta, end) :
        time[j] += 1

for i in range(101) :
    if time[i] == 1 :
        price += A
    elif time[i] == 2:
        price += B
    elif time[i] == 3 :
        price += C

print(price)